src/hb-ot-layout.cc

```diff
@@ -1959,6 +1959,42 @@ hb_ot_layout_substitute_lookup (OT::hb_ot_apply_context_t *c,
 }
 
 #ifndef HB_NO_BASE
+/**
+ * hb_ot_layout_get_baseline_tag_for_script:
+ * @script: a script tag.
+ *
+ * Fetches the dominant baseline tag used by @script.
+ *
+ * Return value: dominant baseline tag for the @script.
+ *
+ * Since: REPLACEME
+ **/
+hb_ot_layout_baseline_tag_t
+hb_ot_layout_get_baseline_tag_for_script (hb_script_t script)
+{
+  switch ((int) script)
+  {
+    /* Unicode-1.1 additions */
+    case HB_SCRIPT_BENGALI:
+    case HB_SCRIPT_DEVANAGARI:
+    case HB_SCRIPT_GURMUKHI:
+    /* Unicode-2.0 additions */
+    case HB_SCRIPT_TIBETAN:
+    /* Unicode-4.1 additions */
+    case HB_SCRIPT_SYLOTI_NAGRI:
+    /* Unicode-5.2 additions */
+    case HB_SCRIPT_MEETEI_MAYEK:
+      return HB_OT_LAYOUT_BASELINE_TAG_HANGING;
+
```

Comment:
Some possible additions: limbu, newa, sharada, tirhuta, siddham, modi, takri, marchen, gunjala gondi, ... probably others, but there are a bunch that I'm not even vaguely familiar with, and don't have fonts readily to hand. (Looking over the unicode charts would be a start...)

And there are a number where it seems hard to know which way to go: e.g. gujarati, where although the letters lack a top line, they're still very reminiscent of the devanagari/bengali/gurmukhi characters, and might expect to be similarly aligned.

Might be worth reaching out to the [w3c i18n group](https://www.w3.org/International/) for input here? My own knowledge, at least, is very fragmentary, and I haven't gone digging for examples of the lesser-known scripts where expectations aren't at all obvious.

Reply:
I don't know. @dscorbett can you make a decision here please?

Reply:
[W3C’s latest draft Indic Layout Requirements](https://www.w3.org/TR/2020/WD-ilreq-20200529/) says that Gujarati has a hanging baseline. Let’s follow that advice.

It also says that Kannada, Telugu, and Odia don’t have a hanging baseline, but its description of “median/mean line” sounds essentially like 'hang' except without a visible headstroke (exactly like Gujarati). The important difference is that, apparently, the bigger font’s median/mean line aligns with the smaller font’s ascent. That’s different enough that these scripts should fall back to 'romn', at least for this first draft. I could definitely believe that many more Brahmic scripts ought to use 'hang' though. I’m just guessing here.

For future reference, Ogham and Mongolian use a central baseline, but there’s not really a tag for that yet. Also, some hieroglyphic and cuneiform scripts should maybe use the ideographic baseline.

Reply:
Interesting examples in there for which characters to synthesize from as well; for those where ka has a cap with an upturn on it that's useful.

Reply:
> Interesting examples in there for which characters to synthesize from as well; for those where ka has a cap with an upturn on it that's useful.

Yeah we tried to do that.